pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_secret_reconcile.go

```diff
@@ -0,0 +1,77 @@
+package clustermanagercontroller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	operatorapiv1 "open-cluster-management.io/api/operator/v1"
+
+	"open-cluster-management.io/ocm/manifests"
+	"open-cluster-management.io/ocm/pkg/operator/helpers"
+)
+
+const (
+	// workDriverConfig is the secret that contains the work driver configurarion
+	workDriverConfig = "work-driver-config"
+)
+
+type secretReconcile struct {
+	operatorKubeClient kubernetes.Interface
+	hubKubeClient      kubernetes.Interface
+	operatorNamespace  string
+	cache              resourceapply.ResourceCache
+	recorder           events.Recorder
+}
+
+func (c *secretReconcile) reconcile(ctx context.Context, cm *operatorapiv1.ClusterManager,
+	config manifests.HubConfig) (*operatorapiv1.ClusterManager, reconcileState, error) {
+	if config.CloudEventsDriverEnabled && config.WorkDriver != string(operatorapiv1.WorkDriverTypeKube) {
+		// check the source secret explicitly as the 'helpers.SyncSecret' doesn't return an error
+		// when the source secret is not found.
+		if _, err := c.operatorKubeClient.CoreV1().Secrets(c.operatorNamespace).Get(ctx,
+			workDriverConfig, metav1.GetOptions{}); errors.IsNotFound(err) {
+			// TODO: set condition if the source secret doesn't exist,
+			return cm, reconcileStop,
```

Comment:
in helpers.SyncSecret, the synced secret will be deleted if the source secret is not found.
so if return here, the synced workDriverConfig secret will not be deleted if workDriverConfig is deleted.
not sure if this is expected ?

Reply:
the source secret should not be deleted per my understanding.

Reply:
can we hav a empty secret slice, and append the "work-driver-config" secret if feature gate is enabled?

Reply:
maybe a secret set instead of a secret slice, to avoid duplicated resync.
done.